domains/page.go

```diff
@@ -0,0 +1,23 @@
+package domains
+
+// ExclusiveStartKeyPagedResponse is object to use when returning database records in paged format using Exclusive Start Key paging
+type ExclusiveStartKeyPagedResponse[Content any, LastEvaluatedKeyType any] struct {
+	// List of records for the response
+	Content []Content `json:"content"`
+	// Pagination details
+	Page ExclusiveStartKeyPage[LastEvaluatedKeyType] `json:"page"`
+}
+
+// ExclusiveStartKeyPage is object to use when limiting and sorting database query results for Exclusive Start Key Paging
+type ExclusiveStartKeyPage[LastEvaluatedKeyType any] struct {
+	// Field by which to order the results
+	OrderByField *string `json:"orderByField"`
+	// Direction in which to order the results ASC/DSC
+	SortDirection *string `json:"sortDirection"`
```

Comment:
as those are optional values, please add `omitempty`
```suggestion
	// Field by which to order the results
	OrderByField *string `json:"orderByField,omitempty"`
	// Direction in which to order the results ASC/DSC
	SortDirection *string `json:"sortDirection,omitempty"`
```